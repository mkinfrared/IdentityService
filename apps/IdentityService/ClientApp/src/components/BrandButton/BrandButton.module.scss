.BrandButton {
  --apple: hsl(0deg 0% 0%);
  --apple-text: hsl(0deg 100% 100%);
  --epic: hsl(0deg 0% 29%);
  --epic-text: hsl(330deg 2% 18%);
  --facebook: hsl(221deg 43% 42%);
  --facebook-text: hsl(0deg 100% 100%);
  --google: hsl(0deg 33% 99%);
  --google-text: hsl(240deg 2% 39%);
  --microsoft: hsl(0deg 0% 98%);
  --microsoft-text: hsl(300deg 0% 45%);
  --steam: hsl(222deg 18% 11%);
  --steam-text: hsl(30deg 3% 71%);
  --vk: hsl(212deg 39% 46%);
  --vk-text: hsl(300deg 100% 100%);

  &.apple {
    --button-spinner-color: var(--apple-text);

    background: var(--apple);
    color: var(--apple-text);

    .brandIcon {
      fill: var(--apple-text);
    }
  }

  &.epic {
    background: var(--epic);
  }

  &.facebook {
    --button-spinner-color: var(--facebook-text);

    background: var(--facebook);
    color: var(--facebook-text);

    .brandIcon {
      border: 1px solid var(--facebook-text);
      border-radius: 0.4rem;
    }
  }

  &.google {
    --button-spinner-color: var(--google-text);

    background: var(--google);
    color: var(--google-text);
  }

  &.microsoft {
    --button-spinner-color: var(--microsoft-text);

    background: var(--microsoft);
    color: var(--microsoft-text);
  }

  &.steam {
    --button-spinner-color: var(--steam-text);

    background: var(--steam);
    color: var(--steam-text);

    .brandIcon {
      path {
        fill: var(--steam-text);
      }
    }
  }

  &.vk {
    background: var(--vk);
    color: var(--vk-text);

    .brandIcon {
      path {
        fill: var(--vk-text);
      }
    }
  }

  .brandIcon {
    fill: initial;
  }
}
