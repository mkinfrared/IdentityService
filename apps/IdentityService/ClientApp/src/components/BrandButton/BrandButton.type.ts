import { ButtonProps } from "@identity-service/ui";

export type BrandButtonProps = ButtonProps & {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  brand: Brand;
};

export enum Brand {
  APPLE = "Apple",
  EPIC_GAMES = "Epic Games",
  FACEBOOK = "Facebook",
  GOOGLE = "Google",
  MICROSOFT = "Microsoft",
  STEAM = "Steam",
  VK = "VKontakte",
}
