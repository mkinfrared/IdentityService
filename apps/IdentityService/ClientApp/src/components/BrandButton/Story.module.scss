.Story {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  max-width: 30rem;
}
