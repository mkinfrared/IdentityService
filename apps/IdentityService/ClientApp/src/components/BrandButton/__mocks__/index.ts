export { default } from "./BrandButton.mock";
