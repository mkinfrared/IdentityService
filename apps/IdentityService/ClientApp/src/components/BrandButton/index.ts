export { default } from "./BrandButton";

export * from "./BrandButton.type";
