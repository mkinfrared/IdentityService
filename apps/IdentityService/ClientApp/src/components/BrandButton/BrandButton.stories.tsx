import { Meta, Story } from "@storybook/react";

import { BrandButton } from "./BrandButton";
import css from "./BrandButton.module.scss";
import { Brand, BrandButtonProps } from "./BrandButton.type";

export default {
  title: "COMPONENTS/BrandButton",
  component: BrandButton,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<BrandButtonProps> = (args) => (
  <div className={css.Story}>
    <BrandButton {...args} brand={Brand.APPLE}>
      Register with Apple
    </BrandButton>
    <BrandButton {...args} brand={Brand.EPIC_GAMES}>
      Register with Epic Games
    </BrandButton>
    <BrandButton {...args} brand={Brand.FACEBOOK}>
      Register with Facebook
    </BrandButton>
    <BrandButton {...args} brand={Brand.GOOGLE}>
      Register with Google
    </BrandButton>
    <BrandButton {...args} brand={Brand.MICROSOFT}>
      Register with Microsoft
    </BrandButton>
    <BrandButton {...args} brand={Brand.STEAM}>
      Register with Steam
    </BrandButton>
    <BrandButton {...args} brand={Brand.VK}>
      Register with VK
    </BrandButton>
  </div>
);

const Default = Template.bind({
  loading: false,
});

Default.args = {
  href: "https://en.wikipedia.org/wiki/Fleetwood_Mac",
};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
