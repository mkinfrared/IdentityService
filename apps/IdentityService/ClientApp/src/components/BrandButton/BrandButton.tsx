import { classNames } from "@identity-service/core";
import { Button } from "@identity-service/ui";
import { FunctionComponent, HTMLAttributes, createElement, memo } from "react";

import { ReactComponent as Apple } from "icons/brands/apple.svg";
import { ReactComponent as Epic } from "icons/brands/epic-games.svg";
import { ReactComponent as Facebook } from "icons/brands/facebook.svg";
import { ReactComponent as Google } from "icons/brands/google.svg";
import { ReactComponent as Microsoft } from "icons/brands/microsoft.svg";
import { ReactComponent as Steam } from "icons/brands/steam.svg";
import { ReactComponent as Vk } from "icons/brands/vk.svg";

import css from "./BrandButton.module.scss";
import { Brand, BrandButtonProps } from "./BrandButton.type";

const brandIcons: Record<
  Brand,
  FunctionComponent<HTMLAttributes<SVGSVGElement>>
> = {
  [Brand.APPLE]: Apple,
  [Brand.EPIC_GAMES]: Epic,
  [Brand.FACEBOOK]: Facebook,
  [Brand.GOOGLE]: Google,
  [Brand.MICROSOFT]: Microsoft,
  [Brand.STEAM]: Steam,
  [Brand.VK]: Vk,
};

const BrandButton = ({ className, brand, ...rest }: BrandButtonProps) => {
  let brandClass: string;

  switch (brand) {
    case Brand.APPLE:
      brandClass = css.apple;

      break;
    case Brand.EPIC_GAMES:
      brandClass = css.epic;

      break;
    case Brand.FACEBOOK:
      brandClass = css.facebook;

      break;
    case Brand.GOOGLE:
      brandClass = css.google;

      break;
    case Brand.MICROSOFT:
      brandClass = css.microsoft;

      break;
    case Brand.STEAM:
      brandClass = css.steam;

      break;
    default:
      brandClass = css.vk;
  }

  const brandIcon = createElement(brandIcons[brand], {
    className: css.brandIcon as string,
  });

  return (
    <Button
      className={classNames(css.BrandButton, brandClass, className)}
      {...rest}
    >
      {brandIcon}
      {brand}
    </Button>
  );
};

export { BrandButton };

export default memo(BrandButton);
