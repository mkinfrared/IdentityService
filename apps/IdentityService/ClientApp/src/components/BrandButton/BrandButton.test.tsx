import { render } from "@testing-library/react";

import { BrandButton } from "./BrandButton";
import { Brand } from "./BrandButton.type";

describe("<BrandButton />", () => {
  const Component = <BrandButton brand={Brand.STEAM} />;

  it("should be defined", () => {
    expect(BrandButton).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("LinkButton");

    expect(element).toBeDefined();
  });

  it("should add 'apple' className when brand is 'Apple'", () => {
    const { getByTestId } = render(<BrandButton brand={Brand.APPLE} />);
    const element = getByTestId("LinkButton");

    expect(element).toHaveClass("apple");
  });

  it("should add 'epic' className when brand is 'Epic Games'", () => {
    const { getByTestId } = render(<BrandButton brand={Brand.EPIC_GAMES} />);
    const element = getByTestId("LinkButton");

    expect(element).toHaveClass("epic");
  });

  it("should add 'facebook' className when brand is 'Facebook'", () => {
    const { getByTestId } = render(<BrandButton brand={Brand.FACEBOOK} />);
    const element = getByTestId("LinkButton");

    expect(element).toHaveClass("facebook");
  });

  it("should add 'google' className when brand is 'Google'", () => {
    const { getByTestId } = render(<BrandButton brand={Brand.GOOGLE} />);
    const element = getByTestId("LinkButton");

    expect(element).toHaveClass("google");
  });

  it("should add 'microsoft' className when brand is 'Microsoft'", () => {
    const { getByTestId } = render(<BrandButton brand={Brand.MICROSOFT} />);
    const element = getByTestId("LinkButton");

    expect(element).toHaveClass("microsoft");
  });

  it("should add 'steam' className when brand is 'Steam'", () => {
    const { getByTestId } = render(<BrandButton brand={Brand.STEAM} />);
    const element = getByTestId("LinkButton");

    expect(element).toHaveClass("steam");
  });

  it("should add 'vk' className when brand is 'Vkontakte'", () => {
    const { getByTestId } = render(<BrandButton brand={Brand.VK} />);
    const element = getByTestId("LinkButton");

    expect(element).toHaveClass("vk");
  });
});
