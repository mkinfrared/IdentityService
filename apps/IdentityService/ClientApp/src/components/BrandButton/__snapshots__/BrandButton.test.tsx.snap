// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrandButton /> should match the snapshot 1`] = `
<div>
  <a
    class="LinkButton-module_LinkButton__x1Ppn BrandButton steam LinkButton-module_opaque__DFeur"
    data-testid="LinkButton"
  >
    <svg
      class="brandIcon"
    >
      steam.svg
    </svg>
  </a>
</div>
`;
